lib/linguist/generated.rb

```diff
@@ -618,6 +619,14 @@ def generated_gimp?
              lines[0].match(/\/\*  GIMP header image file format \([a-zA-Z0-9\- ]+\)\: .+?\.h  \*\//)
     end
 
+    # Internal: Is this a generated Microsoft Visual Studio 6.0 build file?
+    #
+    # Return true or false
+    def generated_visualstudio6?
+      return false unless extname.downcase == '.dsp'
+      lines.any? { |l| l.include? '# Microsoft Developer Studio Generated Build File' }
```

Comment:
Do we really need to look further than the first three lines?

Reply:
Probably not.